src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -1531,6 +1534,18 @@ public static String getLog(Run run) throws IOException {
      * @since 1.607
      */
     public <R extends Run<?,?>> R waitForCompletion(R r) throws InterruptedException {
+        Executor executor = r.getExecutor();
```

Comment:
Whilst this may be technically correct - if we need to look at this then in my opinion there is an in Jenkins core, and we should not work around it in isolation.

we have `isBuilding()` and `isInProgress()` which should be combinable - if this does not work - then the whole ATH is buggy also. (isInProgress()).

Both of these do use the State - (as does isLogUpdated() unless pipeline ooverrides it) so if the log is not being updated, then the build is complete ,

The contract for the state COMPLETE is `Build is completed now, and log file is closed.`  if pipeline is overriing isLogUpdated() then this would surely sound like a break of contract - as COMPLETE is the log is closed - nothing more can add to the log ever...